src/CommonDBTM.php

```
@@ -4330,7 +4330,10 @@ public function filterValues($display = true)
                         case 'email':
                         case 'string':
                         case 'itemlink':
-                            if (is_string($value) && ($length = mb_strlen($value, 'UTF-8')) > 255) {
+                            // Some 'completename' fields may be typed as 'itemlink' (e.g., in CommonTreeDropdown::class).
+                            // These should be excluded from the length check, as they are stored as TEXT or MEDIUMTEXT,
+                            // not VARCHAR. These fields are used solely as item links, not as standard string fields.
+                            if (!is_a(get_called_class(), CommonTreeDropdown::class, true) && is_string($value) && ($length = mb_strlen($value, 'UTF-8')) > 255) {
```

Should not we rely on `completename` field rather than just a class?

---

Should we just remove this truncation ? Now that SQL warnings are handled properly, we would still have a warning triggerred, but at a later moment.

Removing this filtering may prevent detection of unicity in some edge cases, but, in the other handl, will permit to the business rules to be executed using the real input.

Anyway, checking/limiting/truncating the input should never be done, IMHO, with this kind of centralized logic that does not even check the corresponding DB field length. It should be done at the form level, or based on a precise information provided by the class itself, but it should not be done using a `datatype` property that was initially made to describe how the search engine should apply filters and display the corresponding results.

---

> but I would appreciate @cedric-anne opinion on this matter

See my comment posted one minute earlier than yours.

---

I'd agree to remove that truncation; maybe a quick test case should ba deed to make sure we got a proper UI side warning (ca be done from phpunit web tests, not necessary e2e).

---

It does not trigger a SQL warning but fails with a SQL error:
```
glpi.CRITICAL:   *** Uncaught PHP Exception RuntimeException: "MySQL query error: Data too long for column 'email' at row 1 (1406) in SQL query "INSERT INTO `glpi_useremails` (`email`, `users_id`, `is_default`) VALUES ('this-is-a-very-long-email-address-that-should-trigger-a-sql-warning-if-it-is-not-truncated-on-the-php-side-and-it-should-not-be-worse-that-the-current-truncation-logic-that-sometimes-truncates-values-that-would-fit-in-the-db-because-the-column-is-not-limited-to-255-chars@example.com', '5', '1')"." at DBmysql.php line 369
  Backtrace :
  ./src/DBmysql.php:369                              
  ./src/DBmysql.php:1311                             DBmysql->doQuery()
  ./src/CommonDBTM.php:772                           DBmysql->insert()
  ./src/CommonDBTM.php:1388                          CommonDBTM->addToDB()
  ./src/User.php:1779                                CommonDBTM->add()
  ./src/User.php:1337                                User->updateUserEmails()
  ./src/CommonDBTM.php:1822                          User->post_updateItem()
  ./front/user.form.php:142                          CommonDBTM->update()
  ...Glpi/Controller/LegacyFileLoadController.php:59 require()
  ./vendor/symfony/http-kernel/HttpKernel.php:181    Glpi\Controller\LegacyFileLoadController->__invoke()
  ./vendor/symfony/http-kernel/HttpKernel.php:76     Symfony\Component\HttpKernel\HttpKernel->handleRaw()
  ./vendor/symfony/http-kernel/Kernel.php:197        Symfony\Component\HttpKernel\HttpKernel->handle()
  ./public/index.php:56                              Symfony\Component\HttpKernel\Kernel->handle()
``


This patch was applied:
```diff
diff --git a/src/CommonDBTM.php b/src/CommonDBTM.php
index 8a7f017bce..0f80c0aa72 100644
--- a/src/CommonDBTM.php
+++ b/src/CommonDBTM.php
@@ -4419,18 +4419,6 @@ class CommonDBTM extends CommonGLPI
                             }
                             break;
 
-                        case 'email':
-                        case 'string':
-                        case 'itemlink':
-                            if (is_string($value) && ($length = mb_strlen($value, 'UTF-8')) > 255) {
-                                trigger_error(
-                                    "{$value} exceed 255 characters long ({$length}), it will be truncated.",
-                                    E_USER_WARNING
-                                );
-                                $this->input[$key] = mb_substr($value, 0, 255, 'UTF-8');
-                            }
-                            break;
-
                         default:
                             //Plugins can implement their own checks
                             if (!$this->checkSpecificValues($searchOption['datatype'], $value)) {
```

I think it is preferable to have a clear error page rather than a PHP warning that will not be seen by the user, unless it uses the debug mode, and can lead to unexpected side effect. As it may generate some noise for the support team, I would understand if the solution is refused, waiting for a cleaner input validation system.

---

Of course, having an error message can produce extra support questions; but message is clear.

To me, data should not be automatically and silently truncated.